app.js

```diff
@@ -1,13 +1,63 @@
 var express = require('express');
 var app = express();
 var path = require('path');
+var bookshelf = require("./connection.js");
+var bodyParser = require('body-parser');
+var json_parser = bodyParser.json();
+var urlencodedParser = bodyParser.urlencoded({ extended: false });
+
+var Simulators = bookshelf.Model.extend({
+  tableName: "simulators"
+});
 
 app.use(express.static('public'));
 
 app.get('/', function (req, res) {
   res.sendFile(path.join(__dirname + '/public/index.html'));
 });
 
+app.get('/api/simulators', function (req, res) {
+  if (req.query.search) {
+    Simulators.query(function (qb) {
+      qb.where('name', 'LIKE', (req.query.search + '%'))
+    }).orderBy('name', 'ASC')
+      .fetchAll()
+      .then(function (content) {
+        res.json({ content });
+      });
+  }
+  else {
+    Simulators.forge().orderBy('name', 'ASC').fetchAll()
+      .then(function (content) {
+        res.json({content});
+      });
+  }
+
+});
+
+app.post('/api/simulators', json_parser, function (req, res) {
+  //save the database
+  new Simulators({
+    'name': req.body.name,
+    'type_number': req.body.type_number,
+    'price': req.body.price

```

Comment:
Nem kell aposztróf a keyeknek, anélkül is megy. Egy rendes API-nak illik ellenőriznie, amit kap (úgy ahogy most a kliens is ellenőrzi), nem gondolkodás nélkül megpróbálni berakni a db-be...ha valami nem stimmel, akkor hibával kell visszatérni, pl. a response-ban lehet kód, mint error: 'bad_format', és a http status ilyenkor nem 20x, hanem 400...